core/src/tests/addresses_and_ports_functions.cc

```diff
@@ -0,0 +1,113 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+Copyright (C) 2018-2023 Bareos GmbH & Co. KG
+
+This program is Free Software; you can redistribute it and/or
+modify it under the terms of version three of the GNU Affero General Public
+License as published by the Free Software Foundation and included
+in the file LICENSE.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+02110-1301, USA.
+*/
+
+#include "testing_dir_common.h"
+
+#include "lib/address_conf.h"
+#include "lib/bnet.h"
+
+const unsigned short int default_port = 9101;
+
+std::vector<std::string> CreateAddressesFromAddAddress(
+    IPADDR::i_type type,
+    int family,
+    unsigned short default_port,
+    const char* hostname_str,
+    const char* port_str)
+{
+  std::vector<std::string> newaddresses{};
+  char buf[1024];
+  dlist<IPADDR>* addresses = new dlist<IPADDR>();
+
+  dlist<IPADDR>** fake_resource_pointer = &addresses;
+  AddAddress(fake_resource_pointer, type, htons(default_port), family,
+             hostname_str, port_str, buf, sizeof(buf));
+
+  IPADDR* addr = nullptr;
+  foreach_dlist (addr, addresses) {
+    addr->build_address_str(buf, sizeof(buf), true);
+
+    std::string theaddress = std::string(buf);
+    theaddress.pop_back();
+
+    newaddresses.push_back(theaddress);
```

Comment:
This does not matter since this is a test, but you can prevent the extra allocation here by doing
```suggestion
    std::string theaddress = std::string(buf);
    theaddress.pop_back();

    newaddresses.emplace_back(std::move(theaddress));
```
or
```suggestion
    newaddresses.emplace_back(buf).pop_back();
```